kernel/task/src/lib.rs

```diff
@@ -678,13 +678,59 @@ impl Task {
     }
 
     /// Blocks this `Task` by setting its runstate to [`RunState::Blocked`].
-    pub fn block(&self) {
-        self.runstate.store(RunState::Blocked);
+    ///
+    /// Returns the previous runstate. Trying to block an exited or reaped task
+    /// will return an error.
+    pub fn block(&self) -> Result<RunState, ()> {
```

Comment:
1. Should probably return the current runstate on error for extra info
2. Clarify the docs to state that the function only succeeds if the current runstate is `Runnable`
3. Clarify the docs to state that it returns the previous runstate on success and the current runstate on error